source/initial_composition/crust1.cc

```diff
@@ -0,0 +1,225 @@
+                               "is only available for 3d computations."));
+
+      prm.enter_subsection ("Initial composition model");
+      {
+        prm.enter_subsection("Crust1");
+        {
+          data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("Data directory"));
+
+          if ((data_directory.size() > 0) && (data_directory[data_directory.size()-1] != '/'))
+            data_directory += "/";
+
+          crust1_bnds_file_name = prm.get ("Crust1 boundaries file name");
+          crust1_rho_file_name = prm.get ("Crust1 densities file name");
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+
+      const int num_layers = 9;
+
+      const std::string bnds_filename = data_directory+crust1_bnds_file_name;
+
+      // Read CRUST1.0 boundary layer data from file
+      std::istringstream inbnds(Utilities::read_and_distribute_file_content(bnds_filename, this->get_mpi_communicator()));
+
+      std::string bnds_line;
+      for (unsigned int i=0; i<bnds.size(); ++i)
+        {
+          std::getline(inbnds,bnds_line);
+          std::istringstream issbnds(bnds_line);
+          for (unsigned int j=0; j<num_layers; ++j)
+            issbnds >> bnds[i][j];
+        }
+
+      const std::string rho_filename = data_directory+crust1_rho_file_name;
+      
+      // Read CRUST1.0 density data from file
+      std::istringstream inrho(Utilities::read_and_distribute_file_content(rho_filename, this->get_mpi_communicator()));
+      
+      std::string rho_line;
+      for (unsigned int i=0; i<rho.size(); ++i) 
+        { 
+          std::getline(inrho,rho_line);
+          std::istringstream issrho(rho_line);
+          for (unsigned int j=0; j<num_layers; ++j)
+            issrho >> rho[i][j]; 
+        }
+
+      // Fill variable with lat-lon coordinates for CRUST1.0. 
+      unsigned int count = 0;
+      const unsigned int lon_pts_cr1 = 360;
+      const unsigned int lat_pts_cr1 = 180;
+      const double lon_min_cr1 = -179.5;
+      const double lat_min_cr1 = -89.5;
+      for (unsigned int i=0; i<lat_pts_cr1; ++i)
+        {
+          for (unsigned int j=0; j<lon_pts_cr1; ++j)
+            {
+              crust1_lon_lat[count][0] = lon_min_cr1 + double(j);
+              crust1_lon_lat[count][1] = lat_min_cr1 + double(i);
+              count = count + 1;
+             }
+        }
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(Crust1,
+                                              "crust1",
+                                              "A class that implements an initial compositional field containing "
+                                              "densities from the CRUST 1.0 data set. Note that this plugin only "
+                                              "works if there is a compositional field called 'crust1_densities'.")
```

Comment:
Perhaps also worth noting that the crust1.0 model includes data above sea level, which will be cut off if the model doesn't include topography.